packages/services/api/src/modules/audit-logs/providers/audit-log-recorder.ts

```diff
@@ -0,0 +1,77 @@
+import { randomUUID } from 'node:crypto';
+import { Injectable, Scope } from 'graphql-modules';
+import { captureException } from '@sentry/node';
+import { type User } from '../../../shared/entities';
+import { sql as c_sql, ClickHouse } from '../../operations/providers/clickhouse-client';
+import { Logger } from '../../shared/providers/logger';
+import { auditLogSchema, type AuditLogSchemaEvent } from './audit-logs-types';
+
+type AuditLogRecordEvent = AuditLogSchemaEvent & {
+  metadata: {
+    userId: string;
+    userEmail: string;
+    organizationId: string;
+    user: User;
+  };
+};
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+/**
+ * Responsible for recording audit log events and storing them in ClickHouse
+ */
+export class AuditLogRecorder {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogRecorder' });
+  }
+
+  async record(data: AuditLogRecordEvent): Promise<void> {
+    try {
+      const { eventType } = data;
+      const { organizationId, userEmail, userId } = data.metadata;
+      this.logger.debug('Creating audit log event', { eventType });
+
+      const parsedEvent = auditLogSchema.parse(data);
+      const metadata = {
+        user: data.metadata.user,
```

Comment:
This could potentially leak user's password or other things we don't want to expose.
Why? Because it's not parsed, there is no structure, it's only a type definition and typescript that oversees it.